src/gfn/examples/distributions.py

```diff
@@ -0,0 +1,221 @@
+import numpy as np
+import torch
+from torch.distributions import Beta, Categorical, Distribution, MixtureSameFamily
+from torchtyping import TensorType
+
+
+class UnsqueezedCategorical(Categorical):
+    """This class is used to sample actions from a categorical distribution. The samples
+    are unsqueezed to be of shape (batch_size, 1) instead of (batch_size,).
+
+    This is used in `DiscretePFEstimator` and `DiscretePBEstimator`, which in turn are used in
+    `ActionsSampler`."""
+
+    def sample(self, sample_shape=torch.Size()):
+        return super().sample(sample_shape).unsqueeze(-1)
+
+
+class QuarterCircle(Distribution):
+    """Represents distributions on quarter circles (or parts thereof), either the northeastern
+    ones or the southwestern ones, centered at a point in (0, 1)^2. The distributions
+    are Mixture of Beta distributions on the possible angle range.
+
+    Adapted from https://github.com/saleml/continuous-gfn/blob/master/sampling.py
+
+    This is useful for the `Box` environment
+    """
+
+    def __init__(
+        self,
+        delta: float,
+        northeastern: bool,
+        centers: TensorType["n_states", 2],
+        mixture_logits: TensorType["n_states", "n_components"],
+        alpha: TensorType["n_states", "n_components"],
+        beta: TensorType["n_states", "n_components"],
+    ):
+        self.delta = delta
+        self.northeastern = northeastern
+        self.centers = centers
+        self.n_states = centers.shape[0]
+        self.n_components = mixture_logits.shape[1]
+
+        assert mixture_logits.shape == (self.n_states, self.n_components)
+        assert alpha.shape == (self.n_states, self.n_components)
+        assert beta.shape == (self.n_states, self.n_components)
+
+        self.base_dist = MixtureSameFamily(
+            Categorical(logits=mixture_logits),
+            Beta(alpha, beta),
+        )
+
+        self.min_angles, self.max_angles = self.get_min_and_max_angles()
+
+    def get_min_and_max_angles(self):
```

Comment:
General comment - the code is nicely self-documenting but I think it would be good for us to go through at some point and ensure every method has some minimum level of docstring.